src/service/onesignal/index.ts

```diff
@@ -1,22 +1,25 @@
 import {OneSignal} from 'react-native-onesignal';
 
+import getFeatureLoggerInstance, {EFeatureLogFlag} from '../../utils/log/FeatureLog';
 import {getSubscribeableTopic} from '../topics';
 
 type TopicTag = {
   name: string;
 };
 
+const {featLog} = getFeatureLoggerInstance(EFeatureLogFlag.oneSignalUtils);
+
 const rebuildAndSubscribeTags = async () => {
   try {
     const response = await getSubscribeableTopic();
     const {topics = [], history = []} = response?.data || {};
 
-    history?.map((historyItem: TopicTag) => {
-      OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
-      return null;
-    });
-
     return new Promise((resolve) => {
+      history?.map((historyItem: TopicTag) => {
+        OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
+        return null;
+      });
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `map` function is used here for side effects (removing tags), which is not its intended use. The `map` function is meant to create a new array by transforming each element of the original array. For operations that have side effects and don't return meaningful values, it's better to use `forEach`. Also, you are returning a Promise but never resolving or rejecting it. This could lead to potential issues where something is waiting on this promise to settle.
```diff
-      return new Promise((resolve) => {
-       history?.map((historyItem: TopicTag) => {
-         OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
-         return null;
-       });
+      history?.forEach((historyItem: TopicTag) => {
+        OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
+      });

-       setTimeout(() => {
-         topics?.map((topic: TopicTag) => {
-           OneSignal.User.addTag(`better_community_${topic?.name}`, 'true');
+       topics?.forEach((topic: TopicTag) => {
+         OneSignal.User.addTag(`better_community_${topic?.name}`, 'true');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->